### PR TITLE
BaSP-TG-02: Get and Create employee

### PR DESCRIPTION
**Pull request by CristianBucci**
Methods getAll, getById, and post in the entity Employees.

### CODE REVIEW

File: src/validations/employees.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+const validateEmployees = (req, res, next) => {
+  const employeeValidation = Joi.object({
+    name: Joi.string().min(3).max(50).pattern(/^[A-Za-z]+$/)
+      .required(),
+    last_name: Joi.string().min(3).max(50).pattern(/^[A-Za-z]+$/)
+      .required(),
+    phone: Joi.string().min(8).max(15).pattern(/[0-9]/)
+      .required(),
+    email: Joi.string().email({ minDomainSegments: 2, tlds: { allow: ['com', 'net'] } }).required(),
+    password: Joi.string().pattern(/^[a-zA-Z0-9]{3,30}$/).required(),
```
Comment (visojulian) [inline src/validations/employees.js:12]: I think patter should be `/^...{8,50}$/` for the password field

File: src/controllers/employees.js [hunk 2]
```diff
@@ -1,90 +1,87 @@
...
+      message: 'Employees found successfully',
+      data: employees,
+      error: false,
+    });
+  } catch (error) {
+    return res.status(404).json({
+      message: `Employees not found ${error}`,
+      data: undefined,
+      error: true,
+    });
```
[the remark this answers is not in the document]
Comment (agustin-schiaffino) [reply src/controllers/employees.js:18]: I agree, we should use status code 500 on every catch

File: src/validations/employees.js (hunk 1 above)
Comment (agustin-schiaffino) [reply src/validations/employees.js:12]: I agree with this too

File: src/controllers/employees.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (CristianBucci) [reply src/controllers/employees.js:24]: Resolved!

File: src/validations/employees.js (hunk 1 above)
Comment (CristianBucci) [reply src/validations/employees.js]: Resolved!

**Review by agustin-schiaffino — APPROVED** [review]

**Pull request merged**